src/rsa/rsa.rs

```diff
@@ -26,48 +26,18 @@ use std;
 
 use untrusted;
 
+mod padding;
 
-pub struct RSAPadding {
-    digest_alg: &'static digest::Algorithm,
-    digestinfo_prefix: &'static [u8],
-}
-
-#[cfg(feature = "rsa_signing")]
-impl RSAPadding {
-    // Implement padding procedure per EMSA-PKCS1-v1_5,
-    // https://tools.ietf.org/html/rfc3447#section-9.2.
-    fn pad(&self, msg: &[u8], out: &mut [u8]) -> Result<(), ()> {
-        let digest_len =
-            self.digestinfo_prefix.len() + self.digest_alg.output_len;
-
-        // Require at least 8 bytes of padding. Since we disallow keys smaller
-        // than 2048 bits, this should never happen anyway.
-        debug_assert!(out.len() >= digest_len + 11);
-        let pad_len = out.len() - digest_len - 3;
-        out[0] = 0;
-        out[1] = 1;
-        for i in 0..pad_len {
-            out[2 + i] = 0xff;
-        }
-        out[2 + pad_len] = 0;
-
-        let (digest_prefix, digest_dst) = out[3 + pad_len..].split_at_mut(
-            self.digestinfo_prefix.len());
-        digest_prefix.copy_from_slice(self.digestinfo_prefix);
-        digest_dst.copy_from_slice(
-            digest::digest(self.digest_alg, msg).as_ref());
-        Ok(())
-    }
-}
+// `RSA_PKCS1_SHA1` is intentionally not exposed.
+use self::padding::{RSAPadding, RSA_PKCS1_SHA1};
+pub use self::padding::{RSA_PKCS1_SHA256, RSA_PKCS1_SHA384, RSA_PKCS1_SHA512};
```

Comment:
I understand we need this `pub use` to re-export them. But, if we qualify the references to the non-exported other ones using `padding::*` then let's qualify these below too.